File: LATIHAN4DPBO2023/CPP/Dosen.cpp
/*Saya Mochamad Khaairi NIM 2106416 mengerjakan Latihan 4
dalam mata kuliah Desain dan Pemrograman Berorientasi Objek 
untuk keberkahanNya maka saya tidak melakukan kecurangan 
seperti yang telah dispesifikasikan. Aamiin.*/

//include file header
#include "header.hh"

//inisialisasi constructor
Dosen::Dosen() : Sivitas()
{
	nip = "";
	faculty = "";
	major = "";
	lastEdu = "";
	mastery = "";
}

//inisialisasi constructor dengan parameter (menggunakan supper dengan class Sivitas)
Dosen::Dosen(string nik, string name, string gender, string univ, string email, string nip, string faculty, string major, string lastEdu, string mastery) : Sivitas(nik, name, gender, univ, email)
{
	this->nip = nip;
	this->faculty = faculty;
	this->major = major;
	this->lastEdu = lastEdu;
	this->mastery = mastery;
}

//inisialisasi method setter dan getter
void Dosen::setNip(string nip)
{
	this->nip = nip;
}

string Dosen::getNip()
{
	return this->nip;
}

void Dosen::setFaculty(string faculty)
{
	this->faculty = faculty;
}

string Dosen::getFaculty()
{
	return this->faculty;
}

void Dosen::setMajor(string major)
{
	this->major = major;
}

string Dosen::getMajor()
{
	return this->major;
}

void Dosen::setLastEdu(string lastEdu)
{
	this->lastEdu = lastEdu;
}

string Dosen::getLastEdu()
{
	return this->lastEdu;
}

void Dosen::setMastery(string mastery)
{
	this->mastery = mastery;
}

string Dosen::getMastery()
{
	return this->mastery;
}

//inisialisasi destructor
Dosen::~Dosen()
{
}

File: LATIHAN4DPBO2023/Python/Dosen.py
from Sivitas import Sivitas

#inisialisasi class Dosen yang merupakan child dari class Sivitas
class Dosen(Sivitas):
	# atribut
	__nip = ""
	__faculty = ""
	__major = ""
	__lastEdu = ""
	__mastery = ""

	# constructor
	def __init__(self, nik = "", name = "", gender = "", univ = "", email = "", nip = "", faculty = "", major = "", lastEdu = "", mastery = ""):
		super().__init__(nik, name, gender, univ, email)
		self.__nip = nip
		self.__faculty = faculty
		self.__major = major
		self.__lastEdu = lastEdu
		self.__mastery = mastery

	# kumpulan method setter dan getter untuk tiap atribut
	def setNip(self, nip):
		self.__nip = nip

	def getNip(self):
		return self.__nip

	def setFaculty(self, faculty):
		self.__faculty = faculty

	def getFaculty(self):
		return self.__faculty

	def setMajor(self, major):
		self.__major = major

	def getMajor(self):
		return self.__major

	def setLastEdu(self, lastEdu):
		self.__lastEdu = lastEdu

	def getLastEdu(self):
		return self.__lastEdu

	def setMastery(self, mastery):
		self.__mastery = mastery

	def getMastery(self):
		return self.__mastery